pyro/util.py

```diff
@@ -284,7 +290,7 @@ def check_model_guide_match(model_trace, guide_trace, max_plate_nesting=math.inf
         warnings.warn(
             "Found auxiliary vars in the model: {}".format(aux_vars & model_vars)
         )
-    if not (guide_vars <= model_vars | aux_vars):
+    if not (guide_vars <= model_vars | aux_vars | det_vars):
```

Comment:
@fritzo, it looks like deterministic sites are currently treated as (auxiliary) sample sites in the guide when comparing the model to the guide. Do you know if this is the intended behavior? I assumed it wasn't and sketched the following solution.